pkg/runtime/runtime.go

```diff
@@ -0,0 +1,259 @@
+	"k8s.io/client-go/pkg/apis/extensions/v1beta1"
+)
+
+const (
+	python27Http    = "bitnami/kubeless-python@sha256:6789266df0c97333f76e23efd58cf9c7efe24fa3e83b5fc826fd5cc317699b55"
+	python27Pubsub  = "bitnami/kubeless-event-consumer@sha256:5ce469529811acf49c4d20bcd8a675be7aa029b43cf5252a8c9375b170859d83"
+	python34Http    = "bitnami/kubeless-python:test@sha256:686cd28cda5fe7bc6db60fa3e8a9a2c57a5eff6a58e66a60179cc1d3fcf1035b"
+	python34Pubsub  = "bitnami/kubeless-python-event-consumer@sha256:8f92397258836e9c39948814aa5324c29d96ff3624b66dd70fdbad1ce0a1615e"
+	node6Http       = "bitnami/kubeless-nodejs@sha256:b3c7cec77f973bf7a48cbbb8ea5069cacbaee7044683a275c6f78fa248de17b4"
+	node6Pubsub     = "bitnami/kubeless-nodejs-event-consumer@sha256:b027bfef5f99c3be68772155a1feaf1f771ab9a3c7bb49bef2e939d6b766abec"
+	node8Http       = "bitnami/kubeless-nodejs@sha256:1eff2beae6fcc40577ada75624c3e4d3840a854588526cd8616d66f4e889dfe6"
+	node8Pubsub     = "bitnami/kubeless-nodejs-event-consumer@sha256:4d005c9c0b462750d9ab7f1305897e7a01143fe869d3b722ed3330560f9c7fb5"
+	ruby24Http      = "bitnami/kubeless-ruby@sha256:97b18ac36bb3aa9529231ea565b339ec00d2a5225cf7eb010cd5a6188cf72ab5"
+	ruby24Pubsub    = "bitnami/kubeless-ruby-event-consumer@sha256:938a860dbd9b7fb6b4338248a02c92279315c6e42eed0700128b925d3696b606"
+	dotnetcore2Http = "allantargino/kubeless-dotnetcore@sha256:d321dc4b2c420988d98cdaa22c733743e423f57d1153c89c2b99ff0d944e8a63"
+	pubsubFunc      = "PubSub"
+)
+
+type runtimeVersion struct {
+	runtimeID   string
+	version     string
+	httpImage   string
+	pubsubImage string
+}
+
+var python, node, ruby, dotnetcore []runtimeVersion
+var availableRuntimes [][]runtimeVersion
+
+func init() {
+	python27 := runtimeVersion{runtimeID: "python", version: "2.7", httpImage: python27Http, pubsubImage: python27Pubsub}
+	python34 := runtimeVersion{runtimeID: "python", version: "3.4", httpImage: python34Http, pubsubImage: python34Pubsub}
+	python = []runtimeVersion{python27, python34}
+
+	node6 := runtimeVersion{runtimeID: "nodejs", version: "6", httpImage: node6Http, pubsubImage: node6Pubsub}
+	node8 := runtimeVersion{runtimeID: "nodejs", version: "8", httpImage: node8Http, pubsubImage: node8Pubsub}
+	node = []runtimeVersion{node6, node8}
+
+	ruby24 := runtimeVersion{runtimeID: "ruby", version: "2.4", httpImage: ruby24Http, pubsubImage: ruby24Pubsub}
+	ruby = []runtimeVersion{ruby24}
+
+	dotnetcore2 := runtimeVersion{runtimeID: "dotnetcore", version: "2.0", httpImage: dotnetcore2Http, pubsubImage: ""}
+	dotnetcore = []runtimeVersion{dotnetcore2}
+
+	availableRuntimes = [][]runtimeVersion{python, node, ruby, dotnetcore}
+}
+
+// GetRuntimes returns the list of available runtimes as strings
+func GetRuntimes() []string {
+	result := []string{}
+	for _, languages := range availableRuntimes {
+		for _, runtime := range languages {
+			result = append(result, runtime.runtimeID+runtime.version)
+		}
+	}
+	return result
+}
+
+func getAvailableRuntimes(imageType string) []string {
```

Comment:
the method above returns a list of runtime + version (e.g. `python2.7`). In this case it filters depending on the trigger type (if it is `PubSub` it won't return `dotnetcore` since there is not a PubSub image for that).